pennylane/templates/subroutines/commuting_evolution.py

```diff
@@ -125,12 +124,15 @@ def __init__(self, hamiltonian, time, frequencies=None, shifts=None, do_queue=Tr
             self.grad_method = "A"
 
         self.hamiltonian = hamiltonian
-        self.num_params = len(hamiltonian.data) + 1
         self.frequencies = frequencies
         self.shifts = shifts
 
         super().__init__(time, *hamiltonian.data, wires=hamiltonian.wires, do_queue=do_queue, id=id)
 
+    @property
+    def num_params(self):
+        return len(self.hamiltonian.data) + 1

```

Comment:
I am a bit confused about codefactor complaining - shouldn't the property inherit the docstring from `Operator`?

Reply:
Good question. I don't know...

Reply:
This is weird, maybe we should just disable with a pylint comment